vyper/semantics/types/utils.py

```diff
@@ -183,6 +183,13 @@ def get_type_from_annotation(
         )
         return ArrayDefinition(value_type, length, location, is_constant, is_public, is_immutable)
 
+    if isinstance(node, vy_ast.Tuple):
+        values = node.elements
+        values_def: Tuple[BaseTypeDefinition, ...] = ()
+        for v in values:
+            values_def += (get_type_from_annotation(v, DataLocation.UNSET),)
+        return TupleDefinition(values_def)
+
```

Comment:
because this block does not depend at all on the handling of `type_name` earlier in the function and it is just a pure check on the AST node type, i think this should be moved to before the parsing of `type_name`